lambdas/abandon/src/services/abandon-audit-service.ts

```
@@ -0,0 +1,45 @@
+import { EventBridgeClient, PutEventsCommand } from "@aws-sdk/client-eventbridge";
+import { AbandonHandlerConfig } from "../config/abandon-handler-config";
+import { SessionItem } from "../../../common/src/database/types/session-item";
+import { logger } from "../../../common/src/util/logger";
+import { CriError } from "../../../common/src/errors/cri-error";
+
+const eventBridgeClient = new EventBridgeClient({ region: "eu-west-2" });
+
+export function createUserAuditInfo(sessionItem: SessionItem) {
+  const userAuditInfo: Record<string, string> = {
+    govuk_signin_journey_id: sessionItem.clientSessionId,
+    ip_address: sessionItem.clientIpAddress,
+    session_id: sessionItem.sessionId,
+    user_id: sessionItem.subject,
+  };
+  if (sessionItem.persistentSessionId) {
+    userAuditInfo.persistent_session_id = sessionItem.persistentSessionId;
+  }
+  return userAuditInfo;
+}
+
+export async function sendAuditEvent(config: AbandonHandlerConfig, userAuditInfo: any, txmaAuditHeader?: string) {
+  logger.info("Sending audit event ...");
+  const response = await eventBridgeClient.send(
+    new PutEventsCommand({
+      Entries: [
+        {
+          Detail: JSON.stringify({
+            auditPrefix: "IPV_HMRC_RECORD_CHECK_CRI",
+            user: userAuditInfo,
+            deviceInformation: txmaAuditHeader,
+            issuer: config.issuer,
+          }),
+          DetailType: "ABANDONED",
+          EventBusName: config.eventBusName,
+          Source: config.eventBusSource,
+        },
+      ],
+    })
+  );
+  if (response.FailedEntryCount && response.FailedEntryCount > 0) {
+    throw new CriError(500, "Failed to send AuditEvent");
```

Is this the existing behaviour? Other teams are definitely treating failing to send audit event as a silent error because they don't consider TxMA "on the critical path". I'm not sure if we do that in Orange though and happy for us to just keep what was there before

---

Ok yeah I'll just log this, I think that will keep it the same as the existing step function